packages/client/src/App.js

```diff
@@ -0,0 +1,307 @@
+            showModal: false,
+            check7: false,
+            showDraw: false
+        };
+    }
+
+    componentDidMount = async () => {
+        const naver = window.naver;
+        const map = await new naver.maps.Map(
+            d3.select('#map').node(),
+            this.mapOption()
+        );
+
+        this.setState({ map });
+        this.bound = map.getBounds();
+        this.mainPageLoad(map);
+        naver.maps.Event.addListener(map, 'idle', e => {
+            this.bound = map.getBounds();
+            this.mainPageLoad(map);
+            this.DataDelete();
+        });
+
+        const userName = localStorage.getItem('token');
+        if (userName) {
+            const decryptedName = MakeSecret.Decrypt(JSON.parse(userName));
+            this.setState({ name: decryptedName });
+        }
+    };
+
+    handleUserNameOnChange = username => {
+        this.setState({ name: username });
+    };
+
+    mapOption = () => {
+        const naver = window.naver;
+        const mapOptions = {
+            zoomControl: true,
+            zoomControlOptions: {
+                style: naver.maps.ZoomControlStyle.SMALL,
+                position: naver.maps.Position.LEFT_BOTTOM
+            },
+            logoControl: true,
+            logoControlOptions: {
+                position: naver.maps.Position.BOTTOM_RIGHT
+            },
+            scaleControl: true,
+            scaleControlOptions: {
+                position: naver.maps.Position.BOTTOM_RIGHT
+            },
+            mapDataControl: true,
+            mapDataControlOptions: {
+                position: naver.maps.Position.BOTTOM_RIGHT
+            }
+        };
+        return mapOptions;
+    };
+
+    mainPageLoad = map => {
+        const { name, factor } = this.state;
+        const bound = this.bound;
+        axios
+            .post('http://127.0.0.1:3001/user/load', {
+                name,
+                factor,
+                bound
+            })
+            .then(async result => {
+                const data = result.data;
+                const resultData = await data[0];
+                // eslint-disable-next-line no-unused-vars
+                const userData = await data[1];
+                // if there is user data
+
+                if (result.status === 200 || result.status === 201) {
+                    resultData.map(el => {
+                        const { startPos, endPos, zoomLevel } = JSON.parse(
+                            el.figures
+                        );
+                        if (!(el.id in this.drawList)) {
+                            const overlay = new Circle({
+                                position: { startPos, endPos },
+                                naverMap: map,
+                                zoom: zoomLevel
+                            });
+                            overlay.setMap(map);
+                            this.drawList[el.id] = overlay;
+                        }

```

Comment:
이후에 

```suggestion
                        } else {
                            this.drawList[el.id].setMap(null);
                            delete this.drawList[el.id];
                        }
                        
```
이렇게 해야하지 않을까?

Reply:
넵 해당 부분 수정했습니다.